docs/developer.md

```diff
@@ -0,0 +1,123 @@
+
+#### Server mode
+
+We have been building an experimental iodide server based on Python and Django. Currently the main features
+it supports are login/identity (via the github API). To test/run it locally, follow this set of steps:
+
+* Register a [github oauth token](https://github.com/settings/applications/new). Set the homepage URL to be
+"http://localhost:8000" and the authentication callback URL to be "http://localhost:8000/oauth/complete/github/".
+* Copy `.env-dist` to `.env` and set the `GITHUB_CLIENT_ID` and `GITHUB_CLIENT_SECRET` to the values provided above.
+* Make sure you have [Docker](https://docs.docker.com/install/) and [Docker Compose](https://docs.docker.com/compose/install/) installed and working correctly
+* Run `make build && make up`
+* You should now be able to navigate to a test server instance at http://localhost:8000
+
+On subsequent runs, you only need to run `make up`.
+
+Additionally, if you are working on client code, you can run `npm run start` in a separate terminal to run webpack in watch mode (which will make your client code changes visible on page reload). If you require verbose Redux logging, you can set the environment variable `REDUX_LOGGING=VERBOSE` with the command `REDUX_LOGGING=VERBOSE npm run start`
+
+Sometimes, for debugging purposes, it is useful to have a shell session inside the "app" docker container. You
+can use either the `make shell` command (creates a shell session with the "app" user) or the `make root-shell`
+commands (creates a shell session logged in as root, useful for experimenting with new python packages).
+
+### Building the docs
+
+The documentation is written in markdown, and uses
+[mkdocs](https://www.mkdocs.org/) to generate a static website.
+
+To test changes to the docs locally, you'll need to install `mkdocs` and
+`markdown-include`. This is completely independent of the server's Docker image
+described above. If you have `pip` and `python` installed on your system, you
+can install these packages using the command:
+
+```
+pip install mkdocs markdown-include
+```
+
+If that doesn't work, consult the [mkdocs installation
+instructions](https://www.mkdocs.org/#installing-mkdocs).
+
+Then you can run:
+
+```
+mkdocs serve
+```
+
+to preview the docs during development.
+
+To build a local, static copy of the docs, run:
+
+```
+mkdocs build
+```
+
+### Testing
+
+Run `npm test` to run the test suite once, or `npm test --watch` to run the suite in watch mode, which will automatically re-run the tests when the source or tests have changed.
+
+## Contributing
```

Comment:
This section and below feels a bit awkward as part of the "developer documentation" page. Maybe just put it back into README.md for now?